packages/jest-mock/src/index.js

```diff
@@ -642,12 +642,22 @@ class ModuleMockerClass {
   }
 
   spyOn(object: any, methodName: any): any {
+    if (typeof object !== 'object' && typeof object !== 'function') {
+      throw new Error(
+        'Cannot spyOn on a primitive value; ' + this._typeOf(object) + ' given',
+      );
+    }
+
     const original = object[methodName];
 
     if (!this.isMockFunction(original)) {
       if (typeof original !== 'function') {
         throw new Error(
-          'Cannot spyOn the ' + methodName + ' property; it is not a function',
+          'Cannot spy the ' +
+            methodName +
+            ' property because it is not a function; ' +
+            this._typeOf(original) +
+            ' given instead',
         );
```

Comment:
```js
throw new Error(
  'Cannot spyOn on the "' + methodName + '" property because it is not a function\n' +
  printWithType('Got', original, printExpected),
);
```